### DISCUSSION TITLE
ghc "impossible" heap overflow?

### DISCUSSION THREAD

**Discussion by rossc719g** [B-Lang-org/bsc#612] [General]
When compiling bsc on my mac (`make release`) after ebe887730d33208ca9d647d87298699018d47797, I get an error:

```
[180 of 224] Compiling GenABin          ( GenABin.hs, `[redacted]`/bsc/src/comp/../../build/comp/GenABin.o )
ghc-9.2.7: panic! (the 'impossible' happened)
  (GHC version 9.2.7:
	heap overflow

Please report this as a GHC bug:  https://www.haskell.org/ghc/reportabug

make[2]: *** [bsc] Error 1
make[1]: *** [install] Error 2
make: *** [install-src] Error 2
```

Anyone else see this?
I don't think it is actually a memory issue, this machine has 96G..

**Comment by quark17** [comment]
I would guess that this is the issue mentioned in PR #601 (admittedly buried in a rather long thread of comments).  The GHC 9.2 branch has an issue that is not seen with 9.0 (earlier GHC version) or 9.4 or 9.6 (later GHC versions).  I recommend switching to one of those -- we chose to go up to 9.4.7 for the CI and releases.  I boiled it down to a small example and opened an issue in GHC's repo: https://gitlab.haskell.org/ghc/ghc/-/issues/23891

**Comment by rossc719g** [comment]
Thanks!  Upgrading to 9.4.4 fixed it.
